src/platform_impl/linux/mod.rs

```diff
@@ -34,175 +45,199 @@ const BACKEND_PREFERENCE_ENV_VAR: &str = "WINIT_UNIX_BACKEND";
 
 #[derive(Clone)]
 pub struct PlatformSpecificWindowBuilderAttributes {
+    #[cfg(feature = "x11")]
     pub visual_infos: Option<XVisualInfo>,
     pub screen_id: Option<i32>,
     pub resize_increments: Option<Size>,
     pub base_size: Option<Size>,
     pub class: Option<(String, String)>,
     pub override_redirect: bool,
+    #[cfg(feature = "x11")]
     pub x11_window_types: Vec<XWindowType>,
+    #[cfg(feature = "x11")]
     pub gtk_theme_variant: Option<String>,
     pub app_id: Option<String>,
 }
 
 impl Default for PlatformSpecificWindowBuilderAttributes {
     fn default() -> Self {
         Self {
+            #[cfg(feature = "x11")]
             visual_infos: None,
             screen_id: None,
             resize_increments: None,
             base_size: None,
             class: None,
             override_redirect: false,
+            #[cfg(feature = "x11")]
             x11_window_types: vec![XWindowType::Normal],
+            #[cfg(feature = "x11")]
             gtk_theme_variant: None,
             app_id: None,
         }
     }
 }
 
+#[cfg(feature = "x11")]
 lazy_static! {
     pub static ref X11_BACKEND: Mutex<Result<Arc<XConnection>, XNotSupported>> =
         Mutex::new(XConnection::new(Some(x_error_callback)).map(Arc::new));
 }
 
 #[derive(Debug, Clone)]
+#[allow(dead_code)]
 pub enum OsError {
+    #[cfg(feature = "x11")]
     XError(XError),
     XMisc(&'static str),
```

Comment:
`#[cfg]` should be added to the `XMisc` variant, as well. I think that should resolve the `dead_code` warning here.